src/ducks/export/services.spec.js

```diff
@@ -0,0 +1,246 @@
+      },
+      tags: {}
+    },
+    {
+      _id: '0008d7b9134d67cb079d10acc530902f',
+      id: '0008d7b9134d67cb079d10acc530902f',
+      account: {
+        data: accounts.checkings
+      },
+      amount: 78.3,
+      cozyCategoryId: '400840',
+      currency: 'EUR',
+      date: '2021-11-12T12:00:00.000Z',
+      label: 'REMBOURSEMENT FACTURE 0001',
+      originalBankLabel: 'VIREMENT REMBOURSEMENT FACTURE 0001',
+      realisationDate: '2021-11-12T12:00:00.000Z',
+      applicationDate: '2021-10-07T12:00:00.000Z',
+      type: 'transfer',
+      recurrence: {},
+      tags: {
+        data: [tags.melun, tags.vacances, tags.remboursements]
+      }
+    },
+    {
+      _id: '0008d7b9134d67cb079d10acc530902f',
+      id: '0008d7b9134d67cb079d10acc530902f',
+      account: {
+        data: accounts.checkings
+      },
+      amount: -78.3,
+      cozyCategoryId: '400840',
+      currency: 'EUR',
+      date: '2021-10-23T12:00:00.000Z',
+      label: 'TRAVEL AGENCY FACTURE 0001',
+      originalBankLabel: 'TRAVEL AGENCY FACTURE 0001 Card.***4',
+      realisationDate: '2021-10-23T12:00:00.000Z',
+      type: 'credit card',
+      reimbursementStatus: 'reimbursed',
+      recurrence: {},
+      tags: {
+        data: [tags.melun, tags.vacances]
+      }
+    }
+  ]
+
+  return { transactions }
+}
+
+describe('createFormatStream', () => {
+  it('returns a Stream', () => {
+    const stream = createFormatStream()
+
+    expect(stream).toBeInstanceOf(Stream)
+  })
+
+  it('transforms arrays of data into CSV lines', async () => {
+    const data = [
+      '2021-09-10',
+      '2021-09-10',
+      undefined,
+      'GAZ',
+      'PRLV SEPA GAZ',
+      'power',
+      -63,
+      'EUR',
+      undefined,
+      undefined,
+      'Société Générale',
+      'Compte Courant',
+      '00031738274',
+      'Checkings',
+      'Abonnement Gaz',
+      undefined,
+      undefined,
+      undefined,
+      undefined,
+      undefined
+    ]
+
+    const stream = createFormatStream()
+    stream.write(data)
+    stream.end()
+
+    const chunks = []
+    for await (const chunk of stream) {
+      chunks.push(chunk)
+    }
+    const output = Buffer.concat(chunks)
+
+    expect(String(output)).toEqual(
+      [
+        // XXX: The header line is included in the output
+        '"Date";"Realisation date";"Assigned date";"Label";"Original bank label";"Category name";"Amount";"Currency";"Type";"Reimbursement status";"Bank name";"Account name";"Account number";"Account type";"Recurrence name";"Tag 1";"Tag 2";"Tag 3";"Tag 4";"Tag 5"',
+        '"2021-09-10";"2021-09-10";"";"GAZ";"PRLV SEPA GAZ";"power";"-63";"EUR";"";"";"Société Générale";"Compte Courant";"00031738274";"Checkings";"Abonnement Gaz";"";"";"";"";""'
+      ].join('\n')
+    )
+  })
+})
+
+describe('transactionsToCSV', () => {
+  it('returns a generator', () => {
+    const generator = transactionsToCSV([])
+    typeof generator === 'function'
+    typeof generator.next === 'function'
+  })
```

Comment:
Il n'est pas préférable d'avoir au moins un `expect` dans chaque `it` ? 🤔

Reply:
euh si, ce sont des expects qu'il faudrait ici :upside_down_face: .